packages/dynamoose/lib/Model/index.ts

```diff
@@ -443,7 +445,7 @@ export class Model<T extends ItemCarrier = AnyItem> extends InternalPropertiesCl
 			const params: DynamoDB.BatchGetItemInput = {
 				"RequestItems": {
 					[table.getInternalProperties(internalProperties).name]: {
-						"Keys": await Promise.all(keyObjects.map(async (key) => this.Item.objectToDynamo(await this.Item.objectFromSchema(key, this, {"type": "toDynamo", "modifiers": ["set"], "typeCheck": false, "mapAttributes": true}))))
+						"Keys": (await Promise.all(keyObjects)).map((key) => this.Item.objectToDynamo(key))
```

Comment:
the `objectToDynamo` call is left here rather than within `convertKeyToObject` because the `prepareResponse` functions in the batch methods do not work with a "dynamo-ified" object.